streams/src/main/java/org/apache/kafka/streams/state/internals/ValueAndTimestampSerializer.java

```diff
@@ -34,6 +34,51 @@
         timestampSerializer = new LongSerializer();
     }
 
+    private static boolean skipTimestampAndCompareValues(final byte[] left, final byte[] right) {
+        for (int i = Long.BYTES; i < left.length; i++) {
+            if (left[i] != right[i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static long extractTimestamp(final byte[] bytes) {
+        final byte[] timestampBytes = new byte[Long.BYTES];
+        for (int i = 0; i < Long.BYTES; i++) {
+            timestampBytes[i] = bytes[i];
+        }
+        return ByteBuffer.wrap(timestampBytes).getLong();
+    }
+
+    /**
+     * @param left  the serialized byte array of the old record in state store
+     * @param right the serialized byte array of the new record being processed
+     * @return true if the two serialized values are the same (excluding timestamp) or 
+     *              if the timestamp of right is less than left (indicating out of order record)
+     *         false otherwise
+     */
+    public static boolean maskTimestampAndCompareValues(final byte[] left, final byte[] right) {
```

Comment:
Ah, upon reading that last test, I realized that I previously overlooked when you added the timestamp comparison to this method. We should change the method name for maintainability. It no longer just "masks the timestamp and compares the values". Can we instead call it "compareValuesAndCheckForIncreasingTimestamp" or something? I can almost guarantee that one or more people will be badly misled by the current method name.